File: config-auh-core-basic/assets/auth-core-basic-template/src/password/provider/password.repository.ts
import { CrudRepository, Result } from "__SHARED_PACKAGE_NAME__";
import { Password } from "../model/password.entity";

export interface PasswordRepository extends Omit<CrudRepository<Password>, "create"> {
	create(password: Password, userId: string): Promise<Result<Password>>;
	findByUserId(id: string): Promise<Result<Password>>;
}

File: config-auth-core-basic-kt/assets/auth-core-basic-template-kt/src/main/kotlin/com/example/auth/password/repository/PasswordRepository.kt
package com.example.auth.password.repository

import com.example.auth.password.entity.Password

interface PasswordRepository {
    suspend fun create(password: Password): Result<Unit>
    suspend fun findByUserId(userId: String): Password?
    suspend fun findRecentByUserId(userId: String, limit: Int = 5): List<Password>
}
